galaxy_api/auth/models.py

```diff
@@ -1,5 +1,34 @@
+from django.db import models
 from django.contrib.auth.models import AbstractUser
 
+from galaxy_api.common import models as common_models
+
+
+__all__ = (
+    'User',
+    'Tenant',
+)
+
 
 class User(AbstractUser):
+    """Custom user model."""
     pass
+
+
+class Tenant(common_models.TimestampsMixin):
+    """
+    Model representing tenant concept.
+
+    Tenant is a special model that usually represents user or
+    group (organization). It allows configuring objects permissions
```

Comment:
What would be an example of a Tenant? How is it expected to be used?

ie, is it 'admin' and 'users'?   'autohub_users' and 'community_users'?   'rh_sso_users' and 'autohub_local_users' ?

Reply:
It's an object we use for ownership assignment and permission checks. I'll document this with some pictures.
Practically for automation hub it represents an account, so it's one-to-many with users relationships.
For community galaxy it represents a user itself, so it's one-to-one relationship with users, although later if we introduce concept of group or organization in community galaxy it may be extended to handle that.